scene/3d/skeleton_3d.cpp

```diff
@@ -265,11 +265,28 @@ void Skeleton3D::_update_process_order() {
 
 	bones_backup.resize(bones.size());
 
+	_update_bone_names();
+
 	process_order_dirty = false;
 
 	emit_signal("bone_list_changed");
 }
 
+void Skeleton3D::_update_bone_names() {
+	String names;
+	for (int i = 0; i < bones.size(); i++) {
+		if (i > 0) {
+			names += ",";
+		}
+		names += bones[i].name;
+	}
+	concatenated_bone_names = StringName(names);
+}
+
+StringName Skeleton3D::get_concatenated_bone_names() const {
+	return concatenated_bone_names;
```

Comment:
Now here we update them if they changed.
`if (concatenated_bone_names == StringName()) { _update_bone_names(); }`

To allow updating the bone names in a const function, C++ allows declaring `concatenated_bone_names` as mutable. My understanding is mutable is acceptable for caching.